UnityProject/Assets/Scripts/Core/UniversalObjectPhysics.cs

```diff
@@ -0,0 +1,1631 @@
+
+
+	public virtual void Awake()
+	{
+		Collider = this.GetComponent<BoxCollider2D>();
+		floorDecal = this.GetComponent<FloorDecal>();
+		ContextGameObjects[0] = gameObject;
+		defaultInteractionLayerMask = LayerMask.GetMask("Furniture", "Walls", "Windows", "Machines", "Players",
+			"Door Closed",
+			"HiddenWalls", "Objects");
+		attributes = GetComponent<Attributes>();
+		registerTile = GetComponent<RegisterTile>();
+		rotatable = GetComponent<Rotatable>();
+	}
+
+	public Vector3 CalculateLocalPosition()
+	{
+		if (isServer)
+		{
+			return transform.localPosition;
+		}
+		else
+		{
+			if (InitialLocationSynchronised == false)
+			{
+				InitialLocationSynchronised = true;
+				return synchLocalTargetPosition.Vector3;
+			}
+
+			return transform.localPosition;
+		}
+	}
+
+	public void Start()
+	{
+		if (isServer)
+		{
+			SetLocalTarget = new Vector3WithData()
+			{
+				Vector3 = transform.localPosition,
+				ByClient = NetId.Empty
+			};
+		}
+		else
+		{
+			registerTile.ServerSetLocalPosition(synchLocalTargetPosition.Vector3.RoundToInt());
+			registerTile.ClientSetLocalPosition(synchLocalTargetPosition.Vector3.RoundToInt());
+			transform.localPosition = synchLocalTargetPosition.Vector3;
+		}
+
+		if (SnapToGridOnStart && isServer)
+		{
+			transform.position = transform.position.RoundToInt();
+		}
+	}
+
+
+	public override void OnStartClient()
+	{
+		base.OnStartClient();
+	}
+
+	public Size GetSize()
+	{
+		return attributes ? attributes.Size : Size.Huge;
+	}
+
+	public float GetWeight()
+	{
+		return SizeToWeight(GetSize());
+	}
+
+	public virtual void OnEnable()
+	{
+	}
+
+	public virtual void OnDisable()
+	{
+	}
+
+	public struct PullData
+	{
+		public UniversalObjectPhysics NewPulling;
+		public bool WasCausedByClient;
+	}
+
+
+	public struct Vector3WithData
+	{
+		public Vector3 Vector3;
+		public uint ByClient;
+	}
+
+	public void SyncMovementSpeed(float old, float Newmove)
+	{
+		tileMoveSpeed = Newmove;
+	}
+
+
+	public void SyncLocalTarget(Vector3WithData OLDLocalTarget, Vector3WithData NewLocalTarget)
+	{
+		synchLocalTargetPosition = NewLocalTarget;
+		if (isServer) return;
+		if (LocalTargetPosition == NewLocalTarget.Vector3) return;
+		if (isLocalPlayer || PulledBy.HasComponent) return;
+		if (NewLocalTarget.ByClient != NetId.Empty &&  NewLocalTarget.ByClient != NetId.Invalid  && NetworkIdentity.spawned.ContainsKey(NewLocalTarget.ByClient) && NetworkIdentity.spawned[NewLocalTarget.ByClient].gameObject  == PlayerManager.LocalPlayer) return;
+
+		SetLocalTarget = NewLocalTarget;
+
+		if (IsFlyingSliding)
+		{
+			IsFlyingSliding = false;
+			airTime = 0;
+			slideTime = 0;
+			UpdateManager.Remove(CallbackType.UPDATE, FlyingUpdateMe);
+		}
+
+		if (Animating == false && transform.localPosition != NewLocalTarget.Vector3)
+			UpdateManager.Add(CallbackType.UPDATE, AnimationUpdateMe);
+	}
+
+
+	public void StoreTo(ObjectContainer NewParent)
+	{
+		if (NewParent == this) return; //Storing something inside of itself what?
```

Comment:
spawn a blackhole, of course

Reply:
You should totally do this 😂 give some developer a laugh if they make a mistake